Epel template

mcyprian opened the pull request:
Prototype of EPEL template, fixes #25:
- doesn't use modern macros (%py2_install %py2_build)
- forbids base version > 2 if template is used
- disables with_pythonX macro for versions > 2

hroncok commented:
Python 3 EPEL packages are not considered here?

hroncok commented:
Can you include some nice integration test in this PR? So I can see it in action without even trying to use it? :D

mcyprian commented:
I was not able to find complete and clear description of python packaging in EPEL.
[EPEL Packaging wiki page](https://fedoraproject.org/wiki/EPEL:Packaging) says that new macros are not available in EL5 and EL6.
I used an EPEL specfile from package db where Python3 is disabled as an inspiration of this template's design. 

If Python 3 can be used in EPEL packages, can't we simply use [fedora template](https://github.com/fedora-python/pyp2rpm/blob/master/pyp2rpm/templates/fedora.spec) or it's older version [fedora_subidirs.spec](https://github.com/fedora-python/pyp2rpm/blob/master/pyp2rpm/templates/fedora_subdirs.spec) for EPEL packages as well?

Do you think we want something more like spec example on this [wiki page](https://fedoraproject.org/wiki/User:Mstuchli/Python3inEPEL)?

[Here](https://paste.fedoraproject.org/354012/03815151/) is an example of spec file generated running **pyp2rpm coverage -p3 -p35 -s -t epel**. I am not sure if this is what you meant by integration test.

marbu commented:
I'm quite likely to be missing something here, but don't we need a separate template for each EPEL5, EPEL6 and EPEL7? Maybe EPEL5 and EPEL6 are similar enough to share the template, but if I read [Packaging:Python_Old](https://fedoraproject.org/wiki/Packaging:Python_Old) (linked from User:Mstuchli/Python3inEPEL) right, EPEL7 provides significant new features (especially when it comes to python3 support).

hroncok commented:
By integration test I mean a test in the testsuite that would test if this works. I can talk to you in person on Wednesday abut his.

mcyprian commented:
@hroncok: Do you have an idea why some of the tests fails with traceback ImportError: No module named jinja2 despite the fact Jinja2 is on the list of installed deps?

hroncok commented:
No idea

Commit:
Add templates for epel6 and epel7
- Forbid base version > 2 if epel6 template is used
- Disable with_pythonX macro for version > 2

Commit:
Add integration tests, use -vv option in tox test command

Commit:
Fix python3 str decode error

Commit:
Add test of python versions

mcyprian commented:
Integration tests have been fixed. I tested new templates on small set of packages that are directly buildable when fedora.spec template is used. Many packages were built successfully for epel6 and epel7 from the spec files generated using new templates. Some of the packages failed because of missing dependencies in epel repositories. Examples of generated spec files are in files /test/test_data/python-Jinja2_epel6.spec and /test/test_data/python-Jinja2_epel7.spec. 
Do you have some comments/suggestions regarding design of the templates @hroncok @marbu?
